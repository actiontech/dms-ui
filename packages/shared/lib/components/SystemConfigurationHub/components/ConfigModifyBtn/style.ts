import { styled } from '@mui/material';
import { BasicButton } from '../../../BasicButton';

export const SystemConfigModifyBtnStyleWrapper = styled(BasicButton)`
  &.basic-button-wrapper {
    background: ${({ theme }) =>
      theme.sharedTheme.uiToken.colorFillSecondary} !important;
    border: ${({ theme }) =>
      `1px solid ${theme.sharedTheme.uiToken.colorBorderSecondary}`} !important;
    width: 40px !important;
    height: 24px !important;
    margin-top: 3px !important;
    border-radius: 20px !important;
  }
`;
