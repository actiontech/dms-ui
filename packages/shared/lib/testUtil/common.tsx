/* eslint-disable no-console */
export const ignoreAntdFormValidateWarning = () => {
  const warning = console.warn;

  beforeAll(() => {
    console.warn = (...arg) => {
      if (typeof arg[0] === 'string' && arg[0].includes('async-validator:')) {
        return;
      }
      warning(...arg);
    };
  });

  afterAll(() => {
    console.warn = warning;
  });
};

export const ignoreAntdUseFormNotConnectedError = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'Instance created by `useForm` is not connected to any Form element. Forget to pass `form` prop?'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

export const ignoreComponentUncontrolled = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'A component is changing a controlled input to be uncontrolled.'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

// 当为原生的 html 添加大驼峰、小驼峰之类的自定义属性，会出现报错
export const ignoreComponentCustomAttr = () => {
  const error = console.error;
  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase'
        )
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};

/**
 * @description: 在选择 date-range-pick 组件的结束日期 input 元素时，实际选择到了，报此 warning
 */
export const ignoreComponentWarnExpDatePickChooseElement = () => {
  const warn = console.warn;
  beforeAll(() => {
    console.warn = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes(
          'Not match any format. Please help to fire a issue about this.'
        )
      ) {
        return;
      }
      warn(...arg);
    };
  });

  afterAll(() => {
    console.warn = warn;
  });
};

export const ignoreComponentAutoCreatedListNoKey = () => {
  const error = console.error;

  beforeAll(() => {
    console.error = (...arg) => {
      if (
        typeof arg[0] === 'string' &&
        arg[0].includes('Each child in a list should have a unique "key" prop')
      ) {
        return;
      }
      error(...arg);
    };
  });

  afterAll(() => {
    console.error = error;
  });
};
