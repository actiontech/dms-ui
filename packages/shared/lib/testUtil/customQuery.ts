export {
  getBySelector,
  queryBySelector,
  getAllBySelector,
  getSelectContentByFormLabel,
  getSelectValueByFormLabel,
  selectOptionByIndex,
  selectCustomOptionByClassName,
  getValueByInput,
  sleep,
  getHrefByText,
  getAllHrefByText
} from '@actiontech/dms-kit/src/testUtil/customQuery';
