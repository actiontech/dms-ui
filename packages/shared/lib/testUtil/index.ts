export * from './superRender';
export * from './mockHook';
export * from './mockApi';
export * from './customQuery';
export * from './mockRedux';
export * from './common';
export * from './LocationDisplay';
