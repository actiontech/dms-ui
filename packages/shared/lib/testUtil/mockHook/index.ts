export { mockUseCurrentProject } from './mockUseCurrentProject';
export { mockUseCurrentUser } from './mockUseCurrentUser';
export { mockUseDbServiceDriver } from './mockUseDbServiceDriver';
export { mockUsePermission } from './mockUsePermission';
export { mockUseProjectBusinessTips } from './mockUseProjectBusinessTips';
export { mockUseUserInfo } from './mockUseUserInfo';

export * from './data';
