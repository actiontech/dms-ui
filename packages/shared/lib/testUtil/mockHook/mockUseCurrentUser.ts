import * as useCurrentUser from '../../global/useCurrentUser';
import { mockCurrentUserReturn } from './data';

export const mockUseCurrentUser = () => {
  const spy = jest.spyOn(useCurrentUser, 'default');
  spy.mockImplementation(() => mockCurrentUserReturn);
  return spy;
};
