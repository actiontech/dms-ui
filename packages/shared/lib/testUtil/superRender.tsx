import { render, renderHook as renderHookReact } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { ConfigProvider, theme as antdTheme } from 'antd';
import zhCN from 'antd/locale/zh_CN';
import { storeFactory } from './mockRedux';
import { ThemeProvider } from '@mui/system';
import { StyledEngineProvider } from '@mui/material/styles';
import lightTheme from '../theme/light';

export type RenderParams = Parameters<typeof render>;

export type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export interface WrapperOptions {
  routerProps?: MemoryRouterProps;
  initStore?: any;
  recoilRootProps?: RecoilRootPropsCustom;
  theme?: any;
}

export const createTestWrapper = (options?: WrapperOptions) => {
  const {
    routerProps,
    initStore,
    recoilRootProps,
    theme = lightTheme
  } = options || {};

  return ({ children }: { children: React.ReactNode }) => {
    const themeComponent = (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    );

    return (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm, hashed: false }}
      >
        <RecoilRoot {...recoilRootProps}>
          <Provider store={storeFactory(initStore)}>
            <MemoryRouter {...routerProps}>
              <StyledEngineProvider injectFirst>
                {themeComponent}
              </StyledEngineProvider>
            </MemoryRouter>
          </Provider>
        </RecoilRoot>
      </ConfigProvider>
    );
  };
};

export const superRender = (
  ...[ui, option, options]: [...RenderParams, WrapperOptions?]
) => {
  const wrapper = createTestWrapper(options);

  return render(ui, {
    wrapper,
    ...option
  });
};

export const superRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHookReact<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  const wrapper = createTestWrapper(wrapperOptions);

  return renderHookReact(hook, {
    ...renderHookOptions,
    wrapper
  });
};
