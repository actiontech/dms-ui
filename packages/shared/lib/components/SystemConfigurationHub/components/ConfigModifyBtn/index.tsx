import { useTranslation } from 'react-i18next';
import { EditFilled } from '@actiontech/icons';
import BasicToolTips from '../../../BasicToolTips';
import { SystemConfigModifyBtnStyleWrapper } from './style';

const ConfigModifyBtn: React.FC<{ onClick: () => void }> = ({ onClick }) => {
  const { t } = useTranslation();
  return (
    <BasicToolTips title={t('common.modify')} titleWidth={54}>
      <SystemConfigModifyBtnStyleWrapper
        type="text"
        onClick={onClick}
        // todo 临时解决方案，考虑将 theme 全部移至 shared，解决以下问题
        // 1. 由于 theme 独立存在的原因， 所有项目都存在自己的 superRender 方法。
        // 2. 除了 shared 外，所有项目都存在自己的 useThemeStyleData hooks
        icon={<EditFilled color="#292c33" />}
      />
    </BasicToolTips>
  );
};

export default ConfigModifyBtn;
