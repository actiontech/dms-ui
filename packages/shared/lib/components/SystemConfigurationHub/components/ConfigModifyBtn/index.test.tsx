import { cleanup, fireEvent, act } from '@testing-library/react';
import ConfigModifyBtn from '.';
import { mockUseCurrentUser } from '../../../../testUtil/mockHook/mockUseCurrentUser';
import { superRender } from '../../../../testUtil/superRender';
import { getBySelector } from '../../../../testUtil/customQuery';

describe('base/System/components/ConfigModifyBtn', () => {
  beforeEach(() => {
    mockUseCurrentUser();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    cleanup();
  });

  it('render snap', async () => {
    const onClickFn = jest.fn();
    const { baseElement } = superRender(
      <ConfigModifyBtn onClick={onClickFn} />
    );
    const btnEle = getBySelector('.ant-btn', baseElement);

    fireEvent.mouseOver(btnEle);
    await act(async () => jest.advanceTimersByTime(500));
    expect(baseElement).toMatchSnapshot();

    fireEvent.click(btnEle);
    expect(onClickFn).toHaveBeenCalled();
  });
});
