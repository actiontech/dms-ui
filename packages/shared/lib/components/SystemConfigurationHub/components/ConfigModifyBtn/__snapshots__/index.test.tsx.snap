// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/System/components/ConfigModifyBtn render snap 1`] = `
<body>
  <div>
    <div
      class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
    >
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-text ant-btn-icon-only basic-button-wrapper css-11ugcfo"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <svg
              color="#292c33"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/System/components/ConfigModifyBtn render snap 2`] = `
<body>
  <div>
    <div
      class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center ant-tooltip-open basic-tooltips-wrapper css-4jbtl9"
    >
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-text ant-btn-icon-only basic-button-wrapper css-11ugcfo"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <svg
              color="#292c33"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-tooltip ant-zoom-big-fast-appear ant-zoom-big-fast-appear-prepare ant-zoom-big-fast css-dev-only-do-not-override-txh9fw ant-tooltip-placement-top"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-inner"
          role="tooltip"
          style="padding: 0px; border-radius: 10px;"
        >
          <div
            class="css-1lkd6d8"
            width="54"
          >
            修改
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
